app/jobs/course/user_deletion_job.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+class Course::UserDeletionJob < ApplicationJob
+  include TrackableJob
+  queue_as :highest
+
+  protected
+
+  def perform_tracked(course, course_user, current_user)
+    @course = course
+    @course_user = course_user
+    @current_user = current_user
+
+    ActsAsTenant.without_tenant do
+      success = course_user.class.transaction do
+        raise ActiveRecord::Rollback unless course_user.destroy
+
+        true
+      end
+
+      notify_deletion_failure unless success
+    end
+  end
```

Comment:
What happened to resetting the `deleted_at` and `deleter_id` to nil?

Reply:
I just realised when `ActiveRecord::Rollback` is invoked, those 2 values won't be reverted back to `nil` since at this point, `course_user` already has both columns to have some value

I have reverted back the function (it was there, but I deleted it)